### PR TITLE
Add Duet Switch Tester

### PR DESCRIPTION
**Pull request by ai03-2725**
## Description

Adds support for the [Duet switch tester](https://github.com/ai03-2725/duet-switch-tester).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds support for the Duet switch tester

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ai03/duet/config.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 1
+#define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 128
```
Comment (waffle87) [inline keyboards/ai03/duet/config.h:6]:
```suggestion
// Copyright 2023 QMK
// SPDX-License-Identifier: GPL-2.0-or-later

#pragma once

#define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 128
```

File: keyboards/ai03/duet/keymaps/coordinate/README.md
Comment (waffle87) [inline keyboards/ai03/duet/keymaps/coordinate/README.md:1]: Please rename this file to lowercase.

File: keyboards/ai03/duet/keymaps/coordinate/keymap.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
Comment (waffle87) [inline keyboards/ai03/duet/keymaps/coordinate/keymap.c:1]:
```suggestion
// Copyright 2023 QMK
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/ai03/duet/keymaps/default/README.md
Comment (waffle87) [inline keyboards/ai03/duet/keymaps/default/README.md:1]: As above.

File: keyboards/ai03/duet/readme.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Duet Switch Tester
+
+![Cover image](https://raw.githubusercontent.com/ai03-2725/duet-switch-tester/main/images/cover.jpg)
```
Comment (waffle87) [inline keyboards/ai03/duet/readme.md:3]: Please rehost this image on Imgur.

File: keyboards/ai03/duet/rules.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+EEPROM_DRIVER = wear_leveling
+WEAR_LEVELING_DRIVER = rp2040_flash
```
Comment (waffle87) [inline keyboards/ai03/duet/rules.mk:2]:
```suggestion
# This file intentionally left blank
```

File: keyboards/ai03/duet/info.json [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    "matrix_pins": {
+        "cols": ["GP29", "GP28", "GP27", "GP26", "GP25", "GP24", "GP23", "GP22", "GP18", "GP14", "GP13", "GP12", "GP11", "GP10", "GP9", "GP8", "GP7", "GP6"],
+        "rows": ["GP0", "GP1", "GP2", "GP3", "GP4", "GP5"]
+    },
+    "processor": "RP2040",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0028",
+        "vid": "0xA103"
+    },
```
Comment (waffle87) [inline keyboards/ai03/duet/info.json:24]:
```suggestion
    },
    "eeprom": {
        "driver": "wear_leveling"
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/ai03/duet/keymaps/coordinate/keymap.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+	[0] = LAYOUT(
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0
+	),
+
```
Comment (drashna) [inline keyboards/ai03/duet/keymaps/coordinate/keymap.c:19]:
```suggestion
	[0] = LAYOUT(
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0
    ),
```

File: keyboards/ai03/duet/keymaps/coordinate/keymap.c [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+
+/* String generation */
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record)
+{
```
Comment (drashna) [inline keyboards/ai03/duet/keymaps/coordinate/keymap.c:26]:
```suggestion
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

File: keyboards/ai03/duet/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+	[0] = LAYOUT(
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
+		KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0
+	),
+
```
Comment (drashna) [inline keyboards/ai03/duet/keymaps/default/keymap.c:15]:
```suggestion
	[0] = LAYOUT(
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,
        KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0,    KC_0
    ),
```

File: keyboards/ai03/duet/readme.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# Duet Switch Tester
+
+![Cover image](https://i.imgur.com/sFuHJbL.png)
```
Comment (drashna) [inline keyboards/ai03/duet/readme.md:3]:
```suggestion
![Cover image](https://i.imgur.com/sFuHJbLh.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ai03/duet/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT(
```
Comment (waffle87) [inline keyboards/ai03/duet/keymaps/default/keymap.c:7]:
```suggestion
```

File: keyboards/ai03/duet/info.json [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+    "eeprom": {
+        "driver": "wear_leveling"
+    },
```
Comment (waffle87) [inline keyboards/ai03/duet/info.json:27]: This actually isn't needed, sorry.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**